File: Data Structures & Algorithms/count-number-of-islands/submission-1.cpp
class Solution {
public:
int m,n;
void fn(vector<vector<char>>& grid,int i,int j)
{
    if(i<0 || j<0 || i>=m || j>=n || grid[i][j]=='0')
    {
        return;
    }
    grid[i][j]='0';
    fn(grid,i+1,j);
    fn(grid,i-1,j);
    fn(grid,i,j+1);
    fn(grid,i,j-1);
}
    int numIslands(vector<vector<char>>& grid) {
        m=grid.size();
        n=grid[0].size();
        int count=0;
        for(int i=0;i<m;i++)
        {
            for(int j=0;j<n;j++)
            {
                if(grid[i][j]=='1')
                {
                    fn(grid,i,j);
                    count++;
                }
            }
        }
        return count;
    }
};

File: Data Structures & Algorithms/decode-string/submission-1.java
class Solution {
    public String decodeString(String s) {
        Stack<String>stringstack=new Stack<>();
        Stack<Integer>countstack=new Stack<>();
        StringBuilder curr=new StringBuilder("");
        int k=0;
        for(char ch:s.toCharArray())
        {
            if(Character.isDigit(ch))
            {
                k=k*10+(ch-'0');
            }
            else if(ch=='[')
            {
                stringstack.push(curr.toString());
                countstack.push(k);
                curr=new StringBuilder("");
                k=0;
            }
            else if(ch==']')
            {
                String temp=curr.toString();
                curr=new StringBuilder(stringstack.pop());
                int count=countstack.pop();
                for(int i=0;i<count;i++)
                {
                    curr.append(temp);
                }
            }
            else{
                curr.append(ch);
            }
        }
        return(curr.toString());
    }
}